spec/controllers/course/registrations_controller_spec.rb

```diff
@@ -0,0 +1,217 @@
+        subject
+        expect(response).to render_template(:index)
+        expect(assigns(:student_requests)).to contain_exactly(student_request, student_request2)
+        expect(assigns(:staff_requests)).to contain_exactly(staff_request, staff_request2)
+      end
+    end
+  end
+
+  describe '#create' do
+    subject { post :create, course_id: course, registration: student_request.as_json }
+
+    context 'user is signed in' do
+      before do
+        sign_in(user)
+      end
+
+      context 'course is open' do
+        it 'creates a new request after first attempt' do
+          expect { subject }.to change(Course::Registration, :count).by(1)
+          expect(response).to redirect_to(course_path(course))
+          expect(flash[:notice]).
+            to match(I18n.t('course.registrations.create.request_format') %
+                       { role: I18n.t("course.registrations.role.#{student_request.role}") })
+        end
+      end
+
+      context 'course is closed' do
+        before do
+          course.status = :published
+          course.save!
+        end
+        it 'redirects to course page' do
+          subject
+          expect(response).to redirect_to(course_path(course))
+          expect(flash[:error]).to match(I18n.t('course.registrations.create.course_not_open'))
+        end
+      end
+
+      context 'already registered' do
+        before do
+          create(:course_student, course: course, user: user)
+        end
+
+        it 'redirects to course page' do
+          subject
+          expect(response).to redirect_to(course_path(course))
+          expect(flash[:error]).
+            to match(I18n.t('course.registrations.create.already_registered_format') %
+                       { role: I18n.t('course.registrations.role.student') })
+
+        end
+      end
+    end
+
+    context 'user is not signed in' do
+      it 'redirects user to sign in page' do
+        subject
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  describe '#update_multiple' do
+    before do
+      sign_in(create(:administrator))
+    end
+
+    context 'approve selected staff' do
+      subject { post :update_multiple, course_id: course.id, approve_selected: 'staff',
+                     registration_ids: [staff_request.id] }
+
+      it 'creates a course user and destroy the registration' do
+        expect { subject }.to change(CourseUser, :count).by(1).
+                                and change(Course::Registration, :count).by(-1)
+        expect(response).to redirect_to(course_registrations_path(course))
+        expect(flash[:notice]).
+          to match(I18n.t('course.registrations.update_multiple.approve_notice_format') %
+                     { count: 1 })
+      end
+    end
+
+    context 'delete selected staff' do
+      subject { get :update_multiple, course_id: course.id, delete_selected: 'student',
+                    registration_ids: [staff_request.id] }
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Expression at 107, 58 should be on its own line.